connection-loader/src/main/java/com/terracotta/diagnostic/DiagnosticEntityClientService.java

```diff
@@ -52,15 +53,27 @@ public Diagnostics create(final EntityClientEndpoint<EntityMessage, EntityRespon
     return (Diagnostics)Proxy.newProxyInstance(getClass().getClassLoader(), new Class[] {Diagnostics.class},
             new java.lang.reflect.InvocationHandler() {
       @Override
-      public Object invoke(Object proxy, final Method method, Object[] args) throws Throwable {
+      public Object invoke(Object proxy, Method method, final Object[] args) throws Throwable {
         try {
-          return ece.beginInvoke().message(new EntityMessage() {
+          final String methodName = method.getName();
+          InvokeFuture returnValue = ece.beginInvoke().message(new EntityMessage() {
             @Override
             public String toString() {
-              return method.getName();
+              if (methodName.equals("get")) {
+                return "getJMX " + args[0] + " " + args[1];
+              } else if (methodName.equals("set")) {
+                return "setJMX " + args[0] + " " + args[1] + " " + args[2];
+              } else if (methodName.equals("invoke")) {
+                return "invokeJMX " + args[0] + " " + args[1];
+              } else {
+                return methodName;
+              }
             }
-
-          }).invoke().get().toString();
+          }).invoke();
+          // if the server is terminating, never going to get a message back.  just return null
+          if (!methodName.equals("terminateServer") && !methodName.equals("forceTerminateServer")) {
+            return returnValue.get().toString();
```

Comment:
@myronkscott @anthonydahanne : did you talk about the fact that `Diagnostics` method calls should not block and we should be able as a client to use them with timeout, to avoid blocking the health service and user requests ?
We could wrap the InvokeFuture to a Future like it's done in the voltrn proxy and have the `Diagnostics` interface return `Future<Foo>` instead.